### ISSUE TITLE
use own script before OS is loaded....

### ISSUE THREAD

**Issue by codekoch** [maxnet/berryboot#567]
Hi, 
I build a system for schools based on berryboot (http://mediakit.education).
Two questions:
1. Is there any possibility to use a selfmade script (i.e. checking for an update on usb-stick etc.) before the imagefile and the data of the OS are loaded?
2. Is there any command to force a reset of OS image? At the moment I delete the data files within the already started OS, but this is not the optimal solution....

Thx in advance.

**Comment by macmpi** [comment]
Yes, create a [berryboot-init](https://github.com/maxnet/berryboot/blob/berryboot2018/external/package/berrybootgui2/init#L100) sh file and add it to the root of your squash image.
`mksquashfs ./berryboot-init my_image.sqsh`
It will be run as your image is mounted, before actual `switch_root` initiates the OS.

**Comment by codekoch** [comment]
OK. But the problem is that i have to switch off the script sometimes for testing and manipulate the OS. The script is only for updating and resetting the OS.

**Comment by macmpi** [comment]
Maybe [looking at my scripts](https://github.com/macmpi/berryboot-scripts/blob/master/Volumio2/berryboot-init) enabling self-update/reset for some popular distribution images (Volumio2/LibreELEC/Lakka) may help you for what you are trying to achieve.

Quite a hack-ish way to self-copy & reroute the **init** execution with berryboot-init, so that image can then be unmounted, replaced, relaunched...
Would be nice if berryboot could offer a clean mechanism to do such things, but works well so far.

**Comment by codekoch** [comment]
I tried it with a tiny script:

#!/bin/busybox sh
echo "Hello?"
read -n

And i added it to my image with 
mksquashfs ./berryboot-init my_image.img

But nothing happens....no "Hello" appears during boot :-(

I guess I unterstand something wrong... 

Need more help...

**Comment by maxnet** [comment]
It does print "Executing init script" during boot?
If not it did not find your /berryboot-init file.

**Comment by codekoch** [comment]
No, it does'nt print  "Executing init script" . 
But mksquashfs ./berryboot-init my_image.img copied the berryboot-init file....

**Comment by macmpi** [comment]
@codekoch which berryboot version are you using?
I remember some [early september 2018 release](https://github.com/maxnet/berryboot/issues/528) had troubles with locating berryboot-init file. But it has been fixed since (latest Dec 2018 is fine).
